atdgen/src/ocaml.ml

```diff
@@ -134,7 +134,7 @@ let path_of_target (target : target) =
     | Default -> [ "ocaml" ]
     | Biniou -> [ "ocaml_biniou"; "ocaml" ]
     | Json -> [ "ocaml_json"; "ocaml" ]
-    | Bucklescript -> ["ocaml_bs"; "ocaml"]
+    | Melange -> ["ocaml_mel"; "ocaml"]
```

Comment:
I think we want to support the old annotations for backwards compatibility (maybe emitting a warning that the old name will eventually be removed?)

Reply:
I didn't know there was this kind of annotation available. I don't understand well what is the purpose?

They are not documented or tested anywhere in the atd codebase, and at least Ahrefs doesn't use it (afaict), so I am not sure it is worth to spend time providing a smooth migration experience.

Reply:
`ocaml` applies to all generators but `ocaml_xxx` only to `xxx`, and the latter have a priority.

Reply:
Hm, I see. I was expecting atdgen to complain if an unknown generator was used, but i made some tests and observed that doesn't seem to be the case, it just ignores them.

Do you think it's worth to solve the migration for the specific `ocaml_bs` case? Maybe it's more valuable to solve the common case and have atdgen error out if an unknown generator is used?

Reply:
this is what i'm testing fwiw

```
type 'a t <ocaml_foo module="Abs"> = abstract
```

Reply:
I think solving the general case is better. It's better to error out straight away if a tag is not supported, and it will cover the `ocaml_bs` case automatically.

Reply:
Actually now thinking about it, it may be tough since we share generators with the atdgen-www fork.

Reply:
For now, added back the `ocaml_bs` path in https://github.com/ahrefs/atd/pull/375/commits/f94ecc96b39f19ac189cf42469df504b6072d839.